File: b001_fib_seq.cc
// Copyright (c) 2014 liquid_amber
// This file is distributed under MIT license.
// See LICENSE file.

#include <iostream>

int fibonacci(int n) {
  if( n < 2){
    return n;
  }
  return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
  std::cout << fibonacci(40) << "\n";
  return 0;
}

File: b001_fib_seq.rs
// Copyright (c) 2014 liquid_amber
// This file is distributed under MIT license.
// See LICENSE file.

extern crate time;

fn fibonacci(n: int) -> int {
    if n < 2 {
        n
    } else {
        fibonacci(n - 2) + fibonacci(n - 1)
    }
}

fn main() {
    let beg = time::precise_time_s();
    println!("{}", fibonacci(40));
    let end = time::precise_time_s();
    println!("time: {}s", end - beg);
}
